qualtran/bloqs/arithmetic/addition.py

```diff
@@ -266,6 +269,122 @@ def _add_oop_large() -> OutOfPlaceAdder:
 )
 
 
+@frozen
+class SimpleAddConstant(Bloq):
+    r"""Takes |x> to |x + k> for a classical integer `k`.
+
+    Applies addition to input register `|x>` given classical integer 'k'.
+
+    This is the simple version of constant addition because it involves simply converting the
+    classical integer into a quantum parameter and using quantum-quantum addition as opposed to
+    designing a bespoke circuit for constant addition based on the classical parameter.
+
+    Args:
+        bitsize: Number of bits used to represent each integer.
+        k: The classical integer value to be added to x.
+        cvs: A tuple of control variable settings. Each entry specifies whether that
+            control line is a "positive" control (`cv[i]=1`) or a "negative" control `0`.
+        signed: A boolean condition which controls whether the x register holds a value represented
+            in 2's Complement or Unsigned. This affects the ability to add a negative constant.
+
+    Registers:
+        x: A bitsize-sized input register (register x above).
+
+    References:
+        [Improved quantum circuits for elliptic curve discrete logarithms](https://arxiv.org/abs/2001.09580) Fig 2a
+    """
+
+    bitsize: int
+    k: int
+    cvs: Tuple[int, ...] = field(converter=lambda v: (v,) if isinstance(v, int) else tuple(v))
+    signed: bool
+
+    @cached_property
+    def signature(self) -> 'Signature':
+        if len(self.cvs) > 0:
+            return Signature(
+                [Register('ctrl', bitsize=len(self.cvs)), Register('x', bitsize=self.bitsize)]
+            )
+        else:
+            return Signature([Register('x', bitsize=self.bitsize)])
+
+    def on_classical_vals(
+        self, x: 'ClassicalValT', **vars: 'ClassicalValT'
+    ) -> Dict[str, 'ClassicalValT']:
+        if self.cvs:
+            ctrl = vars['ctrl']
+        else:
+            return {'x': x + self.k}
+
+        if np.equal(ctrl, self.cvs):
```

Comment:
In this line, there is an object you wish to be a value but is an array; and another one which is a value but you want it to be an array

`np.equal` returns an array

`ctrl` will be an integer. `Register('x', bitsize=32)` represents a 32-bit integer; `Register('x', bitsize=1, shape=(32,))` represents an array of one-bit bits